azure-pipelines/builds/ci.yml

```diff
@@ -15,24 +15,69 @@ variables:
   - name: Codeql.Enabled
     value: true
 
+  - name: defaultPoolName
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore-Svc-Public' ), False, 'NetCore-Public')]
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore1ESPool-Svc-Internal'), False, 'NetCore1ESPool-Internal')]
+
+  - name: defaultPoolDemandsLinux
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: ImageOverride -equals Build.Ubuntu.1804.Amd64
+
+  - name: defaultPoolDemandsWindows
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: ImageOverride -equals 1es-windows-2022-open
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: ImageOverride -equals 1es-windows-2022
+
 stages:
 - stage: build
   displayName: Build
   jobs:
-  - template: /eng/common/templates/jobs/jobs.yml
-    parameters:
-      enablePublishUsingPipelines: true
-      enablePublishBuildArtifacts: true
-      enablePublishBuildAssets: true
-      publishAssetsImmediately: true
-      artifacts:
-        publish:
-          artifacts: true
-          manifests: true
-      enableSourceBuild: true
-  - template: /azure-pipelines/builds/generatescript-tests.yml
-    parameters:
-      buildConfiguration: Debug
+  - job: Linux
+    pool:
+      name: ${{ variables.defaultPoolName }}
+      demands: ${{ variables.defaultPoolDemandsLinux }}
+    container: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8
```

Comment:
nit: we use mariner in other pipeline, i.e. aspire: https://github.com/dotnet/aspire/blob/74c40701ec6385bcccc2452863ffb3585d36601e/eng/pipelines/azure-pipelines.yml#L63

Reply:
https://github.com/dotnet/arcade/blob/69e73774ddfdacb1eea03e86d0c6ffcdb94eb4dd/eng/common/templates/jobs/source-build.yml#L17

The source-build template uses the centos-stream8 container, which is why I decided to also use it here.